ray_provider/hooks/ray_cli.py

```diff
@@ -0,0 +1,113 @@
+import yaml
+import subprocess
+from pathlib import Path
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+from airflow.hooks.S3_hook import S3Hook
+
+
+from typing import List
+
+
+class RayCliHook(BaseHook):
+    """Simple wrapper around the Ray CLI."""
+
+    def __init__(
+        self,
+        command: str,
+        options: List[str] = None,
+        cluster_config: str = None,
+        script: str = None,
+        script_args: List[str] = None,
+        aws_conn_id: str = "aws_default",
+    ):
+        self.command = command
+        self.options = options
+        self.cluster_config = cluster_config
+        self.script = script
+        self.script_args = script_args
+        self.aws_conn_id = aws_conn_id
+
+        # Make working directory
+        self.ray_dir = Path.cwd() / "ray"
+        self.ray_dir.mkdir()
+        if self.cluster_config:
+            self._get_cluster_config()
+        if script:
+            self._get_script()
+
+        self.supported_cli_cmds = ["up", "down", "submit"]
+
+
+    def _get_cluster_config(self) -> None:
+        if self.cluster_config.startswith("s3://"):
+            self.log.info("Using remote cluster config: %s", self.cluster_config)
+            s3hook = S3Hook(aws_conn_id=self.aws_conn_id)
+            bucket, key = S3Hook.parse_s3_url(self.cluster_config)
+            data = s3hook.read_key(key, bucket_name=bucket)
+            cluster_yaml = yaml.safe_load(data)
+            
+            path = Path(self.cluster_config)
+            self.cluster_config_path = self.ray_dir / path.name
+            with self.cluster_config_path.open("w") as f:
```

Comment:
We should put configs and scripts into temporary directories if we're loading them. I could see it being a security issue to have contexts and scripts lying around after.